packages/config/src/Config.ts

```diff
@@ -381,51 +385,46 @@ const APP_JSON_EXAMPLE = JSON.stringify({
 
 function ensureConfigHasDefaultValues(
   projectRoot: string,
-  exp: ExpoConfig,
+  exp: Partial<ExpoConfig>,
   pkg: JSONObject,
   skipSDKVersionRequirement: boolean = false
 ): { exp: ExpoConfig; pkg: PackageJSONConfig } {
-  if (!exp) exp = {};
-
-  if (!exp.name) {
-    if (typeof pkg.name !== 'string') {
-      pkg.name = path.basename(projectRoot);
```

Comment:
Are you sure that nothing was depending on this side effect? I mean this line was setting the `name` field in the original `pkg` object.

Reply:
I think that's very unlikely, given that `ensureConfigHasDefaultValues` is only called in this same file, always at the end of the function and the original `pkg` object is not returned.